src/checks/substitution_key.rs

```diff
@@ -0,0 +1,101 @@
+use crate::checks::Check;
+use crate::common::*;
+
+pub(crate) struct SubstitutionKeyChecker<'a> {
+    name: &'a str,
+    template: &'a str,
+}
+
+impl Default for SubstitutionKeyChecker<'_> {
+    fn default() -> Self {
+        Self {
+            name: "SubstitutionKey",
+            template: "The {} key isn't properly assigned",
+        }
+    }
+}
+
+impl Check for SubstitutionKeyChecker<'_> {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        let mut value = match line.get_value().map(str::trim) {
+            Some(value) if !value.starts_with('\'') => value,
+            _ => return None,
+        };
+
+        // Checks if keys used in value have both '{' '}' or neither
+        while let Some(index) = value.find('$') {
```

Comment:
We need to make sure it's not an escaped character ( `$`)
Take a look at the [logic for getting substitution keys.](https://github.com/dotenv-linter/dotenv-linter/blob/a603ddab6d4d04a488fc2387478af35a795be1f4/src/common/line_entry.rs#L100)

If a character is escaped we shouldn't consider it  🙂 
Also add tests for these cases please

Reply:
Do you think the latest commit solves this? 😄